packages/virtual-core/src/index.ts

```diff
@@ -276,11 +276,9 @@ export class Virtualizer<TScrollElement = unknown, TItemElement = unknown> {
   }
 
   constructor(opts: VirtualizerOptions<TScrollElement, TItemElement>) {
+    this.scrollRect = opts.initialRect || { width: 0, height: 0 }
```

Comment:
I moved `this.scrollRect` to before the `this.setOptions()` call, because the `this.calculateRange()` call in `setOptions()` otherwise needs an if-guard to prevent a crash on the first init (`this.scrollRect` would not be set on first run).

It being above `this.setOptions()` also means the defaults needed to move to here too. To prevent duplicated defaults (here and in `setOptions()`) I removed `initialRect` and `intialOffset` from the options object altogether, as they are only used in the constructor for initialization anyways. Not sure if that is the best solution, but feel free to object if you'd like to keep `initialRect`/`initialOffset` in the options.